sw/device/lib/dif/dif_rstmgr.c

```diff
@@ -10,84 +10,86 @@
+      kDifRstmgrResetInfo##pub_name == (0x1 << RSTMGR_RESET_##priv_name), \
+      "kDifRstmgrResetInfo" #pub_name " must match the register definition!")
+
+RSTMGR_RESET_INFO_CHECK(Por, INFO_POR);
+RSTMGR_RESET_INFO_CHECK(LowPowerExit, INFO_LOW_POWER_EXIT);
+RSTMGR_RESET_INFO_CHECK(Ndm, INFO_NDM_RESET);
+RSTMGR_RESET_INFO_CHECK(HwReq, INFO_HW_REQ);
+
+_Static_assert(kDifRstmgrResetInfoLast == kDifRstmgrResetInfoHwReq,
+               "Please add `RSTMGR_RESET_INFO_CHECK` for the new reset type!");
+
 _Static_assert(
-    kDifRstmgrCauseSecurityEscalation == RSTMGR_RESET_INFO_CAUSE_ESCALATION,
-    "kDifRstmgrCauseSecurityEscalation must match the register definition!");
+    RSTMGR_PARAM_NUMSWRESETS == 2,
+    "Number of software resets has changed, please update this file!");
+
+// The Reset Manager implementation will have to be updated if the number
+// of software resets grows, as it would span across multiple registers, so
+// there will be multiple of Reset Enable and Reset Control registers. The
+// appropriate offset from the peripheral base would then have to be
+// calculated.
 _Static_assert(
-    kDifRstmgrCauseNonDebugModuleRequest ==
-        RSTMGR_RESET_INFO_CAUSE_NON_DEBUG_MODULE,
-    "kDifRstmgrCauseNonDebugModuleRequest must match the register definition!");
-
-_Static_assert(RSTMGR_SPI_DEVICE_REGEN_SPI_DEVICE_REGEN ==
-                   RSTMGR_USB_REGEN_USB_REGEN,
-               "SPI and USB regen bit offsets are not the same");
-_Static_assert(RSTMGR_RST_SPI_DEVICE_N_RST_SPI_DEVICE_N ==
-                   RSTMGR_RST_USB_N_RST_USB_N,
-               "SPI and USB reset bit offsets are not the same");
-
-#define RSTMGR_REGEN_BIT_OFFSET RSTMGR_SPI_DEVICE_REGEN_SPI_DEVICE_REGEN
-#define RSTMGR_RESET_BIT_OFFSET RSTMGR_RST_SPI_DEVICE_N_RST_SPI_DEVICE_N
-
-static const bitfield_field32_t kRstmgrCausesField = {
-    .mask = RSTMGR_RESET_INFO_CAUSE_MASK,
-    .index = RSTMGR_RESET_INFO_CAUSE_OFFSET,
-};
-
-// Regen registers are "write 1 to clear" = write 1 to lock.
-static const ptrdiff_t kRstmgrRegenOffsetsMap[] = {
-        [kDifRstmgrPeripheralSpi] = RSTMGR_SPI_DEVICE_REGEN_REG_OFFSET,
-        [kDifRstmgrPeripheralUsb] = RSTMGR_USB_REGEN_REG_OFFSET,
-};
-
-// Reset registers, when set to 0, peripheral is held in reset.
-static const ptrdiff_t kRstmgrResetOffsetsMap[] = {
-        [kDifRstmgrPeripheralSpi] = RSTMGR_RST_SPI_DEVICE_N_REG_OFFSET,
-        [kDifRstmgrPeripheralUsb] = RSTMGR_RST_USB_N_REG_OFFSET,
-};
-
-// TODO - static assert to make sure that the above arrays cover all relevant
-//        peripherals?
+    RSTMGR_PARAM_NUMSWRESETS <= 32,
+    "Reset Enable and Control registers span across multiple registers!");
+
+/**
+ * Get Reset Enable and Reset Control register bit index.
+ *
+ * This function calculates the bit position within a register for a specific
+ * peripheral.
+ */
+static bitfield_bit32_index_t rstmgr_regen_rstctrl_bit_index(
```

Comment:
Why do we have this function? Asking because of the static assertion above and the lack of a function that gives the register offset to use if `id` >= 32.

Also, would it make sense to add a bounds check for `peripheral`?